### PR TITLE
feat : TASK-2024-00926 -Document Upload Webpage.

### PR DESCRIPTION
**Pull request by RASNA96**
## Feature description

- Initializes context details for applicants upon validation.
- Authenticates applicant links to ensure valid, non-expired access.
- Updates applicant details, including qualifications, certifications, and employment history.
- Manages file uploads and links attachments to the applicant’s profile.

## Solution description

- Authorization: **authorize_applicant_id** decrypts and validates applicant IDs.
- Applicant Update: **update_register_form** loads and validates form data for updating applicant details.
- File Management: **update_file** handles file uploads, linking files to applicant records if they exist.

## Output screenshots (optional)
Screenshot from 2024-11-13 11-40-30
Screenshot from 2024-11-13 11-40-43
Screenshot from 2024-11-13 11-42-23
Screenshot from 2024-11-13 11-42-29
Screenshot from 2024-11-13 11-43-43
Screenshot from 2024-11-13 11-43-57
Screenshot from 2024-11-13 15-10-53
Screenshot from 2024-11-13 15-10-58
image

## Areas affected and ensured
`beams/www/`

## Is there any existing behavior change of other features due to this code change?
No

## Was this feature tested on the browsers?
  - Chrome

### CODE REVIEW

File: beams/www/job_application_upload/upload_doc.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+			encrypted_applicant_id: encrypted value of HD Ticket ID
+		return HD Ticket ID (example: 123)
+	'''
+	# Set decrypted_ticket_id as false to initialize
+	decrypted_applicant_id = False
+	try:
+		'''
+			Set decrypted_ticket_id as the decripted value (example: 123, the HD Ticket ID)
+			If the encrypted_ticket_id is dirty then decrypt method will raise an exceptions
+		'''
```
Comment (MhmdSinanKT) [inline beams/www/job_application_upload/upload_doc.py:35]: don't add inline documentation like this

File: beams/www/job_application_upload/upload_doc.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+def update_register_form(docname, data):
+    '''
+        Updates a Job Applicant document with form data, handling various sections and child tables.
+        args:
+            docname (str), data (str)
+    '''
+    try:
+        data = json.loads(data)
+        if not data.get("applicant_name"):
+            frappe.throw("Missing required fields: applicant_name")
```
Comment (MhmdSinanKT) [inline beams/www/job_application_upload/upload_doc.py:63]: give labels not field names

File: beams/www/job_application_upload/upload_doc.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                    })
+            for row in data.get("language_proficiency", []):
+                doc.append("language_proficiency", {
+                    "language": row["language"],
+                    "speak": row["speak"],
+                    "read": row["read"],
+                    "write": row["write"]
+                })
+            doc.flags.ignore_validate = True
+            doc.flags.ignore_mandatory = True
```
Comment (MhmdSinanKT) [inline beams/www/job_application_upload/upload_doc.py:118]: is this necessary? if yes make sure ignoring mandatory and validations won't affect the flow

File: beams/www/job_application_upload/upload_doc.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            doc.save(ignore_permissions=True)
+            frappe.db.commit()
```
Comment (MhmdSinanKT) [inline beams/www/job_application_upload/upload_doc.py:120]: don't use unless the code isn't working as intended, can cause unwanted errors

**Review by MhmdSinanKT — CHANGES_REQUESTED** [review]
remove print/console.log statements. Adhere to the code standards.

File: beams/www/job_application_upload/upload_doc.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+            method: "beams.www.job_application_upload.upload_doc.update_register_form",
+            args: {
+                data: JSON.stringify(data),
+                docname: applicant_id
+            },
+            callback: function (r) {
+                alert(r.message === "success" ? 'Job Applicant updated successfully!' : 'Submission completed.');
+            },
+            error: function (err) {
+                console.log("Error:", err);
```
Comment (MhmdSinanKT) [inline beams/www/job_application_upload/upload_doc.js:119]: remove this

File: beams/www/job_application_upload/upload_doc.py [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        filedata_list = filedata["files_data"]
+        if frappe.db.exists(doctype, docname):
+            for filedata_item in filedata_list:
+                filedoc = save_file(filedata_item["filename"], filedata_item["dataurl"], doctype, docname, decode=True, is_private=0)
+                file_name = filedoc.file_url
+        else:
+             for filedata_item in filedata_list:
+                filedoc = save_file(filedata_item["filename"], filedata_item["dataurl"], decode=True, is_private=0)
+                file_name = filedoc.file_url
+    frappe.db.commit()
```
Comment (MhmdSinanKT) [inline beams/www/job_application_upload/upload_doc.py:140]: remove if not necessary

**Review by MhmdSinanKT — CHANGES_REQUESTED** [review]

File: beams/www/job_application_upload/upload_doc.py (hunk 6 above)
Comment (RASNA96) [reply beams/www/job_application_upload/upload_doc.py:140]: yes its needed.

**Review by MhmdSinanKT — APPROVED** [review]

**Pull request merged**